controller/views/partials.py

```diff
@@ -27,11 +27,24 @@ def state_machine(request: HttpRequest) -> HttpResponse:
         else:
             raise ValueError(f"Invalid form: {form.errors}")
 
-    table = tables.FSMTable.from_dict(fsm.get_fsm_architecture(), ci.get_fsm_state())
+    states = fsm.get_fsm_architecture()
+    current_state = ci.get_fsm_state()
+
+    table = tables.FSMTable.from_dict(states, current_state)
+
+    link = 0
+    flowchart = ""
+    for state, events in states.items():
+        for event, target in events.items():
+            flowchart += f"{state}({state}) -->|{event}| {target}({target})\n"
+            if state == current_state:
+                flowchart += f"style {state} fill:#93c54b\n"
+                flowchart += f"linkStyle {link} background-color:#93c54b\n"
+            link += 1
```

Comment:
I'd put this in a separate function, and here just:
```suggestion
    flowchart = make_flowchart(states, current_state)
```

Reply:
Appropriate tests would need to be added.

Reply:
Done.